### PR TITLE
Run go bits inside a container

### PR DESCRIPTION
**Pull request by kargakis**
/cc @deads2k @stevekuznetsov 

Depends on https://github.com/openshift/origin/pull/17681

### CODE REVIEW

File: sjb/config/test_cases/test_branch_origin_end_to_end.yml [hunk 1]
```diff
@@ -24,4 +24,5 @@ extensions:
       script: |-
         OS_BUILD_ENV_PULL_IMAGE=true OS_BUILD_ENV_PRESERVE=_output/local/bin/linux/amd64/end-to-end.test hack/env make build-router-e2e-test
         OS_BUILD_ENV_PULL_IMAGE=true OS_BUILD_ENV_PRESERVE=_output/local/bin/linux/amd64/etcdhelper hack/env make build WHAT=tools/etcdhelper
-        OPENSHIFT_SKIP_BUILD='true' JUNIT_REPORT='true' make test-end-to-end -o build
\ No newline at end of file
+        OS_BUILD_ENV_PULL_IMAGE=true OS_BUILD_ENV_PRESERVE=???? hack/env JUNIT_REPORT='true' make test-end-to-end-unit
```
Comment (kargakis) [inline sjb/config/test_cases/test_branch_origin_end_to_end.yml]: Not sure what artifacts do you need back from the container

**Comment by stevekuznetsov** [comment]
That test requires direct access to the daemon... we've been asking for it to be refactored for what feels like more than a year now. 

/cc @bparees @miminar @dmage

**Comment by bparees** [comment]
> That test requires direct access to the daemon... we've been asking for it to be refactored for what feels like more than a year now.

what test does?

**Comment by bparees** [comment]
(and why the heck do we have a target named `test-end-to-end-unit`? I can't think of anything much more contradictory).

**Comment by stevekuznetsov** [comment]
Hmm I may have tagged the wrong set of people -- I was certain registry tests fell into the bucket (but maybe they just moved out of Origin? I definitely remember them being in here). @knobunc it's currently router tests that expect direct Docker daemon interaction https://github.com/openshift/origin/blob/master/test/end-to-end/router_test.go

**Comment by bparees** [comment]
Yes presumably they moved when the registry was removed. (of course the
same problem may exist for them in their new location)

Ben Parees | OpenShift

**Comment by stevekuznetsov** [comment]
I believe it remains. @mfojtik can we get prioritization for that?

**Pull request closed without merging**